src/iqstat.cpp

```diff
@@ -357,3 +227,116 @@ void usage() {
+        addInxVal(&genQueryInp.sqlCondInp, COL_RULE_EXEC_ID, v2);
+    }
+
+    genQueryInp.maxRows = 50;
+
+    // idempotent version of clearGenQueryOut.
+    auto clear_gq_out = [&o = genQueryOut]{ clearGenQueryOut(o);
+                                            o->attriCnt = 0; };
+    // RAII for both GenQuery variables.
+    auto clean_gq_in_out = [&, &i = genQueryInp, &o = genQueryOut]{ clearGenQueryInp(&i);
+                                                                    clear_gq_out();
+                                                                    free(o); o = nullptr; };
+    // Implement RAII:
+    const auto clean_gq = irods::at_scope_exit{[&] { clean_gq_in_out(); }};
+
+    status = rcGenQuery( Conn, &genQueryInp, &genQueryOut );
+
+    if ( status == CAT_NO_ROWS_FOUND ) {
+        // Need to determine which table "no rows" refers to: USER or RULE.
+        namespace ia = irods::experimental::administration;
+        if (ia::client::exists(*Conn, ia::user{userName})) {
+            if ( allFlag ) {
+                printf( "No delayed rules pending%s\n", diagnostic.c_str() );
+            }
+            else {
+                printf( "No delayed rules pending for user %s%s\n", userName, diagnostic.c_str());
+            }
+            return 0;
+        }
+        else {
+            printf( "User %s does not exist.\n", userName );
+            return 0;
+        }
+    }
+
+    if (brief) {
+        printf( "id     name\n" );
+    }
+
+    static char *columnNames[] = {"id", "name", "rei_file_path", "user_name",
+                                  "address", "time", "frequency", "priority",
+                                  "estimated_exe_time", "notification_addr",
+                                  "last_exe_time", "exec_status"
+                                 };
+
+    auto increment_count_and_print_results = [&] {
+        printGenQueryResults( Conn, status, genQueryOut,
+                              brief ? NULL: columnNames,
+                              int{brief} );
+    };
+
+    increment_count_and_print_results();
+
+    while ( status == 0 && genQueryOut->continueInx > 0 ) {
+        genQueryInp.continueInx = genQueryOut->continueInx;
+        status = rcGenQuery( Conn, &genQueryInp, &genQueryOut );
+        if (!brief && genQueryOut->rowCnt > 0 ) {
+            printf( "----\n" );
+        }
+        increment_count_and_print_results();
+        clear_gq_out();
+    }
+    return 0;
+}
+
+auto genQuery_id_string_to_id(const std::string& key) -> std::int64_t
+{
+    std::string k{key};
+    boost::algorithm::trim(k);
+    try {
+        return std::stoul(k);
+    }
+    catch(const std::invalid_argument&){
+        THROW(SYS_INVALID_INPUT_PARAM, "Delay rule ID has incorrect format.");
+    }
+    catch(const std::out_of_range&){
+        THROW(SYS_INVALID_INPUT_PARAM, "Delay rule ID is too large.");
+    }
+    catch(...){
+        THROW(SYS_INVALID_INPUT_PARAM, "Unknown error parsing delay rule ID.");
```

Comment:
`SYS_UNKNOWN_ERROR`?

Consider, also, a `std::exception` case for the "known" cases.

Reply:
... or is the std::exception case more likely (or guaranteed even) to contain an error string.  That would make sense I guess.  Wll try it that way then.

Reply:
Ok, done ... i think.